src/libaktualizr/uptane/secondaryinterface.h

```diff
@@ -14,6 +14,7 @@ class SecondaryInterface {
   using Ptr = std::shared_ptr<SecondaryInterface>;
 
  public:
+  virtual std::string Type() const = 0;
```

Comment:
I think, this `Type()` as well as the logic of passing around of the `registered` secondaries can be eliminated if we persist required information about IP secondaries within the IP secondary builder/factory, e.g. in `createIPSecondaries` under `if (provision) {` branch. Therefore, we won't need aktualizr instance here and in its `else` we won't need to `secondaries_info.erase(`, we will just load the info from the DB.

Reply:
Passing the `provision` around was indeed needlessly complicated, the last version is a bit more simple.

I'll continue the investigations about the `Type()` method and dedicated storage.